test/credentials_test.dart

```diff
@@ -299,4 +299,185 @@ Future<void> main([List<String>? args]) async {
+  ///    "birthDay": "1999-10-11",
+  ///    "minAge": "10",
+  ///    "maxAge": "90",
+  ///    "company": "Realm",
+  ///    "iat": 1660130537,
+  ///    "exp": 1660349537, //10 years after Aug 2022
+  ///    "aud": "mongodb.com",
+  ///    "iss": "https://realm.io"
+  /// }
+  /// JWT with private key validation is configured in flexible app wich is used by the tests by default.
+  baasTest('JWT validation by specified public key  - login', (configuration) async {
+    final app = App(configuration);
+    String username = "JWT_privatekey_validated_user@realm.io";
+    String userId = "62f394e9bcb9fee0c9aecb76";
+    var token =
+        "eyJraWQiOiIxIiwiYWxnIjoiUlMyNTYiLCJ0eXAiOiJKV1QifQ.eyJzdWIiOiI2MmYzOTRlOWJjYjlmZWUwYzlhZWNiNzYiLCJuYW1lIjp7ImZpcnN0TmFtZSI6IkpvaG4iLCJsYXN0TmFtZSI6IkRvZSJ9LCJlbWFpbCI6IkpXVF9wcml2YXRla2V5X3ZhbGlkYXRlZF91c2VyQHJlYWxtLmlvIiwiZ2VuZGVyIjoibWFsZSIsImJpcnRoRGF5IjoiMTk5OS0xMC0xMSIsIm1pbkFnZSI6IjEwIiwibWF4QWdlIjoiOTAiLCJjb21wYW55IjoiUmVhbG0iLCJpYXQiOjE2NjAxMzA1MzcsImV4cCI6MTY2MDM0OTUzNywiYXVkIjoibW9uZ29kYi5jb20iLCJpc3MiOiJodHRwczovL3JlYWxtLmlvIn0.F9GMR41wIQLwYVjgsFOveIFAaqAOj9rVAAPkRxjrvwNPRyvBGRyqMsf7n6mryLWWRmKKiGiS_yH90W5dCoxHP_rxIErD0yAMU-ZxnzTSFeXXu8fEiPb7U08l_VREQp7zn1EolbOvA6W89vjG9RMblCcmnbJvIVJlKTu9kFUkyI6aHTsaT2Ntd5UhJkXh0PAaQe_Sy9Na7dOfmq9mGHoHhmoNPnCyJB-0sy4mrL7vK4Ks935yI0u8IbwnZrYHEKSr9z474HbSSju6YXpuDJ8LGjqXwnkeIQpDqujIeWCn0iTQvWN7dblr0QFGHurCw-UbHJtX6UK9dvbC85TKg8oTLg";
+    final credentials = Credentials.jwt(token);
+    final user = await app.logIn(credentials);
+
+    expect(user.state, UserState.loggedIn);
+    expect(user.identities[0].id, userId);
+    expect(user.provider, AuthProviderType.jwt);
+    expect(user.profile.email, username);
+    expect(user.profile.name, username);
+    expect(user.profile.gender, "male");
+    expect(user.profile.birthDay, "1999-10-11");
+    expect(user.profile.minAge, "10");
+    expect(user.profile.maxAge, "90");
+    expect(user.profile.firstName, "John");
+    expect(user.profile.lastName, "Doe");
+    expect(user.profile["company"], "Realm");
+  });
+
+  /// JWT Header:
+  /// {
+  ///   "kid": "1",
+  ///   "alg": "RS256",
+  ///   "typ": "JWT"
+  /// }
+  /// JWT Payload data
+  /// {
+  ///    "sub": "62f39187e4b27ad3fb6483ca",  //User.identities.id: If it is a new id the user is created, if it is an existing id the user and profile are updated.
+  ///    "name": {
+  ///      "firstName": "John",
+  ///      "lastName": "Doe"
+  ///    },
+  ///    "email": "JWK_URI_validated_user@realm.io",
+  ///    "gender": "male",
+  ///    "birthDay": "1999-10-11",
+  ///    "minAge": "10",
+  ///    "maxAge": "90",
+  ///    "company": "Realm",
+  ///    "iat": 1660130241,
+  ///    "exp": 1660349241, //10 years after Aug 2022
+  ///    "aud": "mongodb.com",
+  ///    "iss": "https://realm.io"
+  /// }
+  /// JWKS url validation is configured in autoConfirm app.
+  baasTest('JWT validation by using JWK URI - login', (configuration) async {
```

Comment:
This test is identical to the one above and is testing server functionality - I don't think we need to have it. If the test above passes, this means that the token is correctly passed from dart to core to the server. What the server does with the token and how it validates it against the public key or against jwk uri should be irrelevant for us.

Reply:
Shall I remove everything related to JWK url?

Reply:
JWK url is removed.